nautobot/ipam/views.py

```diff
@@ -661,6 +673,107 @@ def post(self, request):
         )
 
 
+class IPAddressMergeView(view_mixins.GetReturnURLMixin, view_mixins.ObjectPermissionRequiredMixin, View):
+    queryset = IPAddress.objects.all()
+
+    def get_required_permission(self):
+        return get_permission_for_model(self.queryset.model, "change")
+
+    def get(self, request):
+        host_values = (
+            self.queryset.values("host").order_by("host").annotate(count=models.Count("host")).filter(count__gt=1)
+        )
+        if host_values:
+            item = host_values[0]
+            queryset = self.queryset.filter(host__in=[item["host"]])
+            return render(
+                request=request,
+                template_name="ipam/ipaddress_merge.html",
+                context={
+                    "queryset": queryset,
+                    "return_url": self.get_return_url(request),
+                },
+            )
+        else:
+            msg = "No duplicate IPs found."
+            messages.warning(request, msg)
+            return redirect(self.get_return_url(request))
+
+    def post(self, request):
+        logger = logging.getLogger(__name__)
+        collapsed_ips = IPAddress.objects.filter(pk__in=request.POST.getlist("pk"))
+        merged_attributes = request.POST
+        if collapsed_ips and "_skip" not in request.POST:
+            with transaction.atomic():
+                namespace = Namespace.objects.get(pk=merged_attributes.get("namespace"))
+                status = Status.objects.get(pk=merged_attributes.get("status"))
+                if merged_attributes.get("tenant"):
+                    tenant = Tenant.objects.get(pk=merged_attributes.get("tenant"))
+                else:
+                    tenant = None
+                if merged_attributes.get("role"):
+                    role = Role.objects.get(pk=merged_attributes.get("role"))
+                else:
+                    role = None
+                if merged_attributes.get("tags"):
+                    tag_pk_list = merged_attributes.get("tags").split(",")
+                    tags = Tag.objects.filter(pk__in=tag_pk_list)
+                else:
+                    tags = []
+                if merged_attributes.get("nat_inside"):
+                    nat_inside = IPAddress.objects.get(pk=merged_attributes.get("nat_inside"))
```

Comment:
This may be a tricky one to support and might even make sense to exclude for now as I think about it - there's a good chance the "inside" NAT IP is also duplicated? In general I think at the very least you'd need to check for a `MultipleObjectsReturned` exception.

Reply:
`merged_attributes.get("nat_inside")` contains a UUID string that guarantees to return a single object. I am not sure I understand why we need a `MultipleObjectsReturned` exception.